addons/mail/models/res_partner.py

```diff
@@ -100,6 +105,9 @@ def find_or_create(self, email, assert_valid_email=False):
         create_values = {self._rec_name: parsed_name or parsed_email}
         if parsed_email:  # otherwise keep default_email in context
             create_values['email'] = parsed_email
+        if default_create_values_getter:
+            create_values.update({k: v for k, v in default_create_values_getter(parsed_email).items()
+                                  if k in self._fields})
```

Comment:
It’s not supposed to so it is a good idea to remove it indeed as it could hide errors. I have removed the checks and updated the dictionary directly with the values returned, letting the ORM check whether the keys are fields or not.